types/react-redux/index.d.ts

```diff
@@ -35,6 +36,18 @@ import {
 
 import hoistNonReactStatics = require('hoist-non-react-statics');
 
+/**
+ * This interface can be augmented by users to add default types for the root state when
+ * using `react-redux`.
+ * Use module augmentation to append your own type definition in a your_custom_type.d.ts file.
+ * https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation
+ */
+// tslint:disable-next-line:no-empty-interface
+export interface ConfigurableRootState {}
+
+export type AnyIfEmpty<T extends object> = keyof T extends never ? any : T;
+export type SafeRootState = AnyIfEmpty<ConfigurableRootState>;
```

Comment:
Making `SafeRootState` `any` instead of `{}` as a default can make certain errors harder to spot